### PR TITLE
Allow explicitly specifying additional block navigation menu options in block edit function

### PR DESCRIPTION
**Pull request by adamziel**
## Description

This PR is a Proof of Concept of explicitly specifying per-block ellipsis menu items in the edit.js file.

This is a part of a sequence of few PRs.  I propose merging/reviewing these PRs in the following order:

1. ~~#22427 - introduce very basic ellipsis menu to the block navigation~~
1. #22463 - allow the block to define additional ellipsis menu items
1. #22462 - introduce an abstraction that takes care of the toolbar items and ellipsis menu items at the same time

## How has this been tested?
1. Go to the experimental navigation screen
1. Open the ellipsis menu
1. Confirm the Add Submenu is there

Note that adding a submenu transfer focus to the Navigation menu page - this is something I am looking forward to addressing in a separate PR once the slot mechanics from this PR are in place.

## Screenshots

Zrzut ekranu 2020-05-21 o 11 34 14

## Types of changes
New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Comment by adamziel** [comment]
ATM it only renders the additional menu items if the block is selected. Let's explore ways around that once we discuss the direction.

File: packages/block-editor/src/components/block-navigation/block.js [hunk 1]
```diff
@@ -22,6 +22,7 @@ import DescenderLines from './descender-lines';
 import BlockNavigationBlockContents from './block-contents';
 import BlockSettingsDropdown from '../block-settings-menu/block-settings-dropdown';
 import { useBlockNavigationContext } from './context';
+import EllipsisMenuControls from './ellipsis-menu-controls';
```
Comment (draganescu) [inline packages/block-editor/src/components/block-navigation/block.js:25]: I wonder is if this name is OK 😁 `EllipsisMenuControls` should maybe become `BlockSettingsMenuControls` and this `BlockSettingsMenu` name to be propagated. `EllipsisMenu` sounds like the menu of some ellipsis and the real name of the menu is Block Settings.

File: packages/block-editor/src/components/block-settings-menu/block-settings-dropdown.js [hunk 2]
```diff
@@ -145,6 +145,7 @@ export function BlockSettingsDropdown( { clientIds, ...props } ) {
 									/>
 								) }
 							</MenuGroup>
+							{ children }
 							<BlockSettingsMenuControls.Slot
```
Comment (draganescu) [inline packages/block-editor/src/components/block-settings-menu/block-settings-dropdown.js:149]: I don't understand why are we adding another slot and can't use the BlockSettingsMenuControls API to add new items in this menu?

Comment (adamziel) [reply packages/block-editor/src/components/block-settings-menu/block-settings-dropdown.js:149]: @draganescu see my comment here: https://github.com/WordPress/gutenberg/pull/22427#discussion_r428552883

File: packages/block-editor/src/components/block-navigation/block.js (hunk 1 above)
Comment (adamziel) [reply packages/block-editor/src/components/block-navigation/block.js:25]: Agreed, the name ellipsis menu is... not fantastic. On the other hand the menu we want to implement here should be a superset of `BlockSettingsMenu`. See the comment here: https://github.com/WordPress/gutenberg/pull/22427#discussion_r428552883

**Comment by draganescu** [comment]
I replied to [#22427 (comment)](https://github.com/WordPress/gutenberg/pull/22427#discussion_r428552883) as well, but will iterate here, if we're reusing the stuff that is in the Block Settings menu, that has it's own slots, can't we reuse that and not add another slot/fill?

I think what we do in this navigator will find its way to the rest of the editor, and in that context I expect that what is in the block's ellipsis menu is also in the navigator item's ellipsis menu, as the item is a smaller representation of the block itself in the hierarchy. 

If this menu has a different slot/fill there will be things stuck here but not in blocks :)

**Comment by adamziel** [comment]
> I replied to #22427 (comment) as well, but will iterate here, if we're reusing the stuff that is in the Block Settings menu, that has it's own slots, can't we reuse that and not add another slot/fill?

If we did reuse that, then both the navigation menu AND the toolbar menu would contain the extra options. I introduced a new slot to make sure this does not happen. If you think it would make sense, then let's discuss it in the original issue: https://github.com/WordPress/gutenberg/issues/22089

> If this menu has a different slot/fill there will be things stuck here but not in blocks :)

That was my intention :-)

**Comment by adamziel** [comment]
Let's use the approach from https://github.com/WordPress/gutenberg/pull/22674 instead

**Pull request closed without merging**